src/firebolt/model/engine.py

```diff
@@ -320,6 +325,95 @@ def stop(
 
         return engine
 
+    def update(
+        self,
+        name: Optional[str] = None,
+        engine_type: Optional[EngineType] = None,
+        scale: Optional[int] = None,
+        spec: Optional[str] = None,
+        auto_stop: Optional[int] = None,
+        warmup: Optional[WarmupMethod] = None,
+        description: Optional[str] = None,
+    ) -> Engine:
+        """
+        update the engine, and returns an updated version of the engine, all parameters
+        could be set to None, this would keep the old engine parameter value
+        """
+
+        class _EngineUpdateRequest(FireboltBaseModel):
+            """Helper model for sending Engine update requests."""
+
+            account_id: str
+            desired_revision: Optional[EngineRevision]
+            engine: Engine
+            engine_id: str
+            update_mask: FieldMask
+
+        # Update the engine parameters
+        self.name = name if name else self.name
+        self.description = description
+
+        # Update engine settings
+        engine_settings_params = {
+            "engine_type": engine_type,
+            "auto_stop_delay_duration": f"{auto_stop * 60}s" if auto_stop else None,
+            "warm_up": warmup,
+        }
+        self.settings = EngineSettings.default(**prune_dict(engine_settings_params))
+
+        # Update the engine desired_revision if needed
+        desired_revision = None
+        if (scale or spec) and self.latest_revision_key:
+            rm = self._service.resource_manager
+            desired_revision = rm.engine_revisions.get_by_key(self.latest_revision_key)
+
+            if spec:
+                instance_type_key = rm.instance_types.get_by_name(
+                    instance_type_name=spec,
+                    region_name=rm.regions.regions_by_key[self.compute_region_key].name,
+                ).key
+
+                desired_revision.specification.db_compute_instances_type_key = (
+                    instance_type_key
+                )
+                desired_revision.specification.proxy_instances_type_key = (
+                    instance_type_key
+                )
+
+            if scale:
+                desired_revision.specification.db_compute_instances_count = scale
+
+        # Send the update request
+        response = self._service.client.patch(
+            url=ACCOUNT_ENGINE_URL.format(
+                engine_id=self.engine_id, account_id=self._service.account_id
+            ),
+            headers={"Content-type": "application/json"},
+            json=_EngineUpdateRequest(
+                account_id=self._service.account_id,
+                engine=self,
+                desired_revision=desired_revision,
+                engine_id=self.engine_id,
+                update_mask=FieldMask(
+                    paths=list(
```

Comment:
nit: Can we calculate paths in a separate function, or at least take it out from patch request?